docs/manual/cluster-admin/how-to-renew-k8s-cert.md

```diff
@@ -0,0 +1,129 @@
+sudo mv admin.conf /etc/kubernetes/
+sudo mv controller-manager.conf /etc/kubernetes/
+sudo mv kubelet.conf /etc/kubernetes/
+sudo mv scheduler.conf /etc/kubernetes/
+
+# Restart the master components
+sudo kill -s SIGHUP $(pidof kube-apiserver)
+sudo kill -s SIGHUP $(pidof kube-controller-manager)
+sudo kill -s SIGHUP $(pidof kube-scheduler)
+
+# Verify master component certificates - should all be 1 year in the future
+# Cert from api-server
+echo -n | openssl s_client -connect localhost:6443 2>&1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout | grep Not
+# Cert from controller manager
+echo -n | openssl s_client -connect localhost:10257 2>&1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout | grep Not
+# Cert from scheduler
+echo -n | openssl s_client -connect localhost:10259 2>&1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout | grep Not
+```
+
+### Generate new kubelet certicates
+
+On the master node use the following commands generate the new kubelet.conf file:
+
+```bash
+sudo kubeadm alpha kubeconfig user --org system:nodes --client-name system:node:$(hostname) > kubelet.conf
+sudo chown root:root kubelet.conf
+sudo chmod 600 kubelet.conf
+
+# Drain
+kubectl drain --ignore-daemonsets $(hostname)
+# Stop kubelet
+sudo systemctl stop kubelet
+# Delete files
+sudo rm /var/lib/kubelet/pki/*
+# Copy file
+sudo mv kubelet.conf /etc/kubernetes/
+# Restart
+sudo systemctl start kubelet
+# Uncordon
+kubectl uncordon $(hostname)
+
+# Check kubelet
+echo -n | openssl s_client -connect localhost:10250 2>&1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout | grep Not
+```
+
+### Generate a new token for the worker node kubelets
+
+On the master node use the following commands generate the new token:
+
+```bash
+sudo kubeadm token create
+```
+
+### Update certs in worker nodes
+
+Use a playbook to do update the certs in batch. Create a file named `renew-worker-certs.yaml` and replace `<The generated token in above step>`:
+
+```yaml
+---
+- hosts: all
+  tasks:
+    - name: join k8s
+      shell: |
+        systemctl stop kubelet
+        rm /etc/kubernetes/kubelet.conf
+        rm /var/lib/kubelet/pki/*
+        sed -i "s/token: .*/token: <The generated token in above step>/" /etc/kubernetes/bootstrap-kubelet.conf
+        systemctl start kubelet
+```
+
+If you don't have the `hosts.yml` file, please run the commands in OpenPAI source code to generate one:
+
+```bash
+contrib/kubespray/script/k8s_generator.py -l layout.yaml -c config.yaml -o <output_folder>
+```
+
+And run the following commands:
+
+```bash
+ansible-playbook -i hosts.yml --limit '!prodheads000001' --become --become-user root renew-worker-cert.yaml
+```
+
+### Remove token after all workers have been updated
+
+The token will expire in 24h if we don't do this step.
+
+```bash
+sudo kubeadm token delete TOKEN-FROM-CREATION-ON-MASTER
+```
```

Comment:
This command should be run from the master node?

Reply:
Add `On master node` comments to this command.